src/recipes/Calendar.types.ts

```diff
@@ -0,0 +1,90 @@
+import { DispatchAction } from "./DatePicker.types";
+
+export type DayOfWeek = 1 | 2 | 3 | 4 | 5 | 6 | 7;
+
+export type DayData = {
+  dayOfMonth: number;
+  isInCurrentMonth: boolean;
+  available: boolean;
+  selected: boolean;
+  today: boolean;
+  date: Date;
+};
+
+export type WeekData = {
```

Comment:
```suggestion
export type WeekData = Record<DayOfWeek, DayData>
```

Although I find these sorts of shapes great candidates for inferring from a concrete object or Array. This means you don't have to create the types, the types will match the actual implementation.